### PR TITLE
chore: same stamp unit test

### PR DESCRIPTION
**Pull request by esadakar**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/1961)

### CODE REVIEW

**Review by zelig — APPROVED** [review]
> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/1961)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @acud and @aloknerurkar)

File: pkg/localstore/mode_put.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ func (db *DB) putRequest(batch *leveldb.Batch, binIDs map[uint8]uint64, item she
 		// if a chunk is found with the same postage stamp index,
 		// replace it with the new one only if timestamp is later
 		if !later(previous, item) {
-			return false, 0, err
+			return false, 0, nil
```
Comment (acud) [inline pkg/localstore/mode_put.go:216]: I wonder if we need to have an explicit error here. @zelig?

File: pkg/localstore/mode_put.go [hunk 2]
```diff
@@ -274,6 +274,26 @@ func (db *DB) putUpload(batch *leveldb.Batch, binIDs map[uint8]uint64, item shed
 		return true, 0, nil
 	}
 
+	previous, err := db.postageIndexIndex.Get(item)
+	if err != nil {
+		if !errors.Is(err, leveldb.ErrNotFound) {
```
Comment (acud) [inline pkg/localstore/mode_put.go:279]: reduce nesting by `if err != nil && !errors.Is(err,leveldb.ErrNotFound)`

File: pkg/localstore/mode_put_test.go [hunk 3]
```diff
@@ -484,87 +485,114 @@ func TestModePut_sameChunk(t *testing.T) {
...
-					t.Fatalf("expected %v, got %v", storage.ErrNotFound, err)
-				}
-			})
-		}
+	stamp := postagetesting.MustNewStamp()
+	ts := time.Now().Unix()
+
+	for _, tcn := range []struct {
+		name         string
+		mode         storage.ModePut
```
Comment (acud) [inline pkg/localstore/mode_put_test.go:496]: if test cases are the same for all modes, then `mode` can be factored out to an enclosing `for _,mode := range []storage.ModePut{storage.ModePutSync,storage.ModePutRequest.....} { for ..... }`

File: pkg/localstore/mode_put.go (hunk 2 above)
Comment (esadakar) [reply pkg/localstore/mode_put.go:279]: then the next if statement will need to recheck if `err == nil`. I do like the explicit nesting here :)

Comment (acud) [reply pkg/localstore/mode_put.go:279]: fair enough :)

File: pkg/localstore/mode_put.go (hunk 1 above)
Comment (zelig) [reply pkg/localstore/mode_put.go:216]: no error needed

File: pkg/localstore/mode_put_test.go [hunk 4]
```diff
@@ -484,79 +485,56 @@ func TestModePut_sameChunk(t *testing.T) {
...
-				_, err = db.Get(ctx, storage.ModeGetLookup, first.Address())
-				if !errors.Is(storage.ErrNotFound, err) {
-					t.Fatalf("expected %v, got %v", storage.ErrNotFound, err)
-				}
-			})
-			t.Run("chunk on same index - timestamps in reverse order", func(t *testing.T) {
+	stamp := postagetesting.MustNewStamp()
+	ts := time.Now().Unix()
+
+	for _, modeTc := range []storage.ModePut{storage.ModePutRequest, storage.ModePutRequestPin, storage.ModePutSync} {
```
Comment (aloknerurkar) [inline pkg/localstore/mode_put_test.go:494]: In the previous tests we were using mixed modes in the tests as well. (mode1 and mode2 could be same or different)

Are these adding any value? I guess if the handling for different modes changes in the future this test might be required?

File: pkg/localstore/mode_put_test.go [hunk 5]
```diff
@@ -484,79 +485,56 @@ func TestModePut_sameChunk(t *testing.T) {
...
 				newItemsCountTest(db.pullIndex, 1)(t)
-				_, err = db.Get(ctx, storage.ModeGetLookup, first.Address())
+
+				_, err = db.Get(ctx, storage.ModeGetLookup, tc.persistChunk.Address())
 				if err != nil {
 					t.Fatalf("expected no error, got %v", err)
 				}
-				_, err = db.Get(ctx, storage.ModeGetLookup, second.Address())
+				_, err = db.Get(ctx, storage.ModeGetLookup, tc.discardChunk.Address())
 				if !errors.Is(storage.ErrNotFound, err) {
```
Comment (acud) [inline pkg/localstore/mode_put_test.go:538]: `err, storage.ErrNotFound`

**Review by aloknerurkar — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**